state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2000,7 +2020,7 @@ impl<T: Config<I>, I: 'static> DepositApi<T::TargetChain> for Pallet<T, I> {
 	> {
 		if let Some(refund_params) = &refund_params {
 			ensure!(
-				refund_params.retry_duration <= MAX_SWAP_RETRY_DURATION_BLOCKS,
+				refund_params.retry_duration <= MaxSwapRetryDurationBlocks::<T, I>::get(),
```

Comment:
Can be done in a separate issue - but since we can have retry_duration set to 0, I think this is a good case to test on its own. That if it's 0, we still attempt to swap and then refund if it fails.

Reply:
PRO-1511